modules/gib.js

```diff
@@ -0,0 +1,217 @@
+    // The "tygem" flag affects the positioning.
+
+    let points = []
+
+    if (boardsize < 4) {
+        return points
+    }
+
+    if (handicap > 9) {
+        handicap = 9
+    }
+
+    let d
+    if (boardsize < 13) {
+        d = 2
+    } else {
+        d = 3
+    }
+
+    if (handicap >= 2) {
+        points.push([boardsize - d, 1 + d])
+        points.push([1 + d, boardsize - d])
+    }
+
+    // Experiments suggest Tygem puts its 3rd handicap stone in the top left
+
+    if (handicap >= 3) {
+        if (tygem) {
+            points.push([1 + d, 1 + d])
+        } else {
+            points.push([boardsize - d, boardsize - d])
+        }
+    }
+
+    if (handicap >= 4) {
+        if (tygem) {
+            points.push([boardsize - d, boardsize - d])
+        } else {
+            points.push([1 + d, 1 + d])
+        }
+    }
+
+    if (boardsize % 2 == 0) {      // No handicap > 4 on even sided boards
+        return points
+    }
+
+    let mid = (boardsize + 1) / 2
+
+    if (handicap === 5 || handicap === 7 || handicap === 9) {
+        points.push([mid, mid])
+    }
+
+    if (handicap >= 6) {
+        points.push([1 + d, mid])
+        points.push([boardsize - d, mid])
+    }
+
+    if (handicap >= 8) {
+        points.add([mid, 1 + d])
+        points.add([mid, boardsize - d])
+    }
+
+    return points
+}
+
+exports.parse = function (input) {
+
+    // Try UTF-8 encoding... in any case it's better than binary.
+
+    input = iconv.decode(Buffer.from(input, 'binary'), 'utf8')
+
+    let lines = input.split('\n')
+
+    let tree = gametree.new()
+    let root = {}
+    tree.nodes.push(root)
+
+    root.FF = ['4']
+    root.GM = ['1']
+    root.SZ = ['19']
+
+    let node = root
+
+    for (let n = 0; n < lines.length; n++) {
+
+        let line = lines[n].trim()
+
+        if (line.startsWith('\\[GAMEBLACKNAME=') && line.endsWith('\\]')) {
+
+            let s = line.slice(16, -2)
+            root.PB = [s]
+
+        } else if (line.startsWith('\\[GAMEWHITENAME=') && line.endsWith('\\]')) {
+
+            let s = line.slice(16, -2)
+            root.PW = [s]
+
+        } else if (line.startsWith('\\[GAMECONDITION=')) {
+
+            // Hard-coded the common komi cases.
+            // For better results, we could do a regex here instead.
+
+            if (line.toLowerCase().includes('black 6.5 dum')) {
+                root.KM = ['6.5']
+            } else if (line.toLowerCase().includes('black 7.5 dum')) {
+                root.KM = ['7.5']
+            } else if (line.toLowerCase().includes('black 0.5 dum')) {
+                root.KM = ['0.5']
+            }
+
+        } else if (line.startsWith('\\[GAMERESULT=')) {
+
+            let score = null
+            let strings = line.split(' ')
+
+            // Try to find score by assuming any float found is the score.
+
+            for (let i in strings) {
+                try {
+                    let p = parseFloat(strings[i])  // Does this ever throw?
```

Comment:
No, both `parseFloat` and `parseInt` do not throw in Javascript, so the `try` block is unnecessary. But they may return `NaN`.